File: CinemaBookingApp/Front-end/CinemaBookingApp/src/types/room.ts
import {RoomTypeInRoomProps} from './roomType';
import {ShowingTimeInRoomProps} from './showingTime';

export interface RoomInCinemaProps {
  roomId: number;
  name: string;
  roomType: RoomTypeInRoomProps;
  showingTimes: ShowingTimeInRoomProps[];
}

File: CinemaBookingApp/CinemaBookingApp/CinemaBookingApp/Models/Entities/Room.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace CinemaBookingApp.Models.Entities
{
    public class Room
    {
        [Key]
        public long RoomId { get; set; }

        [Required]
        [StringLength(100)]
        public string Name { get; set; }

        // Foreign keys
        public long RoomTypeId { get; set; }
        public long CinemaId { get; set; }

        // Navigation properties
        [ForeignKey("RoomTypeId")]
        public RoomType RoomType { get; set; }

        [ForeignKey("CinemaId")]
        public Cinema Cinema { get; set; }

        public ICollection<Seat> Seats { get; set; }
        public ICollection<ShowingTime> ShowingTimes { get; set; }
    }
}
